files/firstLaunch.js

```diff
@@ -207,6 +214,129 @@ module.exports = async () => {
+        })
+
+        speedtest.stderr.setEncoding("utf8")
+        speedtest.stderr.on("data", data => {
+            console.log('There was an error performing the speedtest, skipping...')
+            chooseRenderingType()
+        })
+    }
+
+    async function downloadLibrespeedCli() {
+        // set different links for different platforms (windows, linux, mac)
+        let link;
+        const platform = process.platform;
+        if (platform === "win32") {
+            link = "https://github.com/librespeed/speedtest-cli/releases/download/v1.0.10/librespeed-cli_1.0.10_windows_amd64.zip"
+        } else if (platform === "linux") {
+            link = "https://github.com/librespeed/speedtest-cli/releases/download/v1.0.10/librespeed-cli_1.0.10_linux_amd64.tar.gz"
+        }
+
+        // make directory for speedtest-cli if it doesn't exist
+        if (!fs.existsSync(`${process.cwd()}/files/librespeed-cli`)) {
+            fs.mkdirSync(`${process.cwd()}/files/librespeed-cli`)
+        }
+
+        // Prompt user to run speedtest
+        let { doSpeedtest } = await inquirer.prompt({
+            name: "doSpeedtest",
+            type: "list",
+            message: "Should we run a speedtest to o!rdr? If you chose Manually Download before, choose it again.",
+            choices: ['Automatically download', 'Manually download', 'Don\'t run'],
+            default: false
+        })
+
+        if (doSpeedtest === 'Don\'t run') return chooseRenderingType();
+
+
+        // download librespeed-cli
+        if (doSpeedtest === 'Automatically download') {
+            if (fs.existsSync(`${process.cwd()}/files/librespeed-cli/librespeed-cli` + (platform === "win32" ? ".exe" : ""))) {
+                console.log("Librespeed-cli already exists.")
+                return runSpeedtest()
+            }
+
+            const output = `${process.cwd()}/files/librespeed-cli/librespeed-cli.zip`
+            let download = wget.download(link, output)
+    
+            download.on("error", err => {
+                console.log("There was an error downloading librespeed-cli.")
+            })
+    
+            download.on("start", fileSize => {
+                console.log(`Downloading librespeed-cli at ${link}: ${fileSize} bytes to download...`)
+            })
+    
+            download.on("end", () => {
+                console.log(`Finished downloading librespeed-cli.`)
+                
+                // unzip librespeed-cli
+                console.log(`Unzipping librespeed-cli...`)
+                fs.createReadStream(`${process.cwd()}/files/librespeed-cli/librespeed-cli.zip`)
+                .pipe(unzipper.Extract({ path: `${process.cwd()}/files/librespeed-cli/` }))
+                
+                // when unzipping is done, delete zip file
+                .on("close", () => {
+                    console.log(`Finished unzipping librespeed-cli.`)
+                    fs.unlinkSync(`${process.cwd()}/files/librespeed-cli/librespeed-cli.zip`)
```

Comment:
after unzipping and deleting the zip we need to chmod on linux
`if (process.platform === "linux") fs.chmodSync("files/librespeed-cli/librespeed-cli", "755")`

Reply:
Completely forgot about that lmao, nice catch